api/v1alpha08/sonataflowplatform_services_types.go

```diff
@@ -32,61 +32,8 @@ type ServiceSpec struct {
 	Enabled *bool `json:"enabled,omitempty"`
 	// Persists service to a datasource of choice. Ephemeral by default.
 	// +optional
-	Persistence *PersistenceOptions `json:"persistence,omitempty"`
+	Persistence *PersistencePlatformSpec `json:"persistence,omitempty"`
 	// PodTemplate describes the deployment details of this platform service instance.
 	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="podTemplate"
 	PodTemplate PodTemplateSpec `json:"podTemplate,omitempty"`
 }
-
-// PersistenceOptions configure the services to persist to a datasource of choice
-// +kubebuilder:validation:MaxProperties=1
```

Comment:
it's ok to rename this but this CRD schema validation config should be brought over to the new struct. this is critical going forward as additional persistence types are added. a user should only be allowed to select one at a given time.

Reply:
Added back.